corporate/lib/stripe.py

```diff
@@ -1621,6 +1645,14 @@ def generate_stripe_invoice(
                 invoice_period_start, CustomerPlan.FIXED_PRICE_PLAN_DURATION_MONTHS
             )
 
+        if on_free_trial and billing_modality == "send_invoice":
+            # Paid plan starts at the end of free trial.
+            invoice_period_start = invoice_period_end
+            purchased_months = 1
+            if billing_schedule == CustomerPlan.BILLING_SCHEDULE_ANNUAL:
+                purchased_months = 12
+            invoice_period_end = add_months(invoice_period_end, purchased_months)
```

Comment:
Should this be using `start_of_next_billing_cycle`?

Reply:
That function needs a `plan`. But yeah, it should. Added to my TODO do to that as a prep commit for next PR, where I fix this along with:
* Latency issue with `invoice_plan` event being generate late when calling `generate-stripe-fixtures` resulting in errors.
* Complete `nocoverage` added in this PR`.

Reply:
Sounds good!